modules/model_loader.py

```diff
@@ -20,7 +20,11 @@ def load_file_from_url(
         file_name = os.path.basename(parts.path)
     cached_file = os.path.abspath(os.path.join(model_dir, file_name))
     if not os.path.exists(cached_file):
-        print(f'Downloading: "{url}" to {cached_file}\n')
-        from torch.hub import download_url_to_file
-        download_url_to_file(url, cached_file, progress=progress)
-    return cached_file
+        try:
+            print(f'Downloading: "{url}" to {cached_file}\n')
+            from torch.hub import download_url_to_file
+            download_url_to_file(url, cached_file, progress=progress)
+        except Exception as e:
+            print(f"Failed to download \"{url}\" to {cached_file}, reason : {e}")
```

Comment:
@codezeros downloading an invalid URL still reports as successful to the frontend despite raising an exception.
This has to be fixed (reverted).

`Failed to download "invalid url" to /Volumes/data/private/development/ai/fooocus/models/checkpoints/invalid url, reason : unknown url type: 'invalid url'
`

![Screenshot 2024-03-09 at 16 19 39](https://github.com/lllyasviel/Fooocus/assets/9307310/0b08a872-5cc0-44bd-84a5-0d19a53bfae6)

Reply:
@mashb1t  thanks for notifying the problem i will look into it and also making some improvements based on previous feedback (in progress)